libs/community/langchain_community/vectorstores/clickhouse.py

```diff
@@ -205,10 +197,53 @@ def __init__(
         )
         # Enable JSON type
         self.client.command("SET allow_experimental_object_type=1")
-        # Enable index
-        self.client.command(f"SET allow_experimental_{self.config.index_type}_index=1")
+        if self.config.index_type:
+            # Enable index
+            self.client.command(
+                f"SET allow_experimental_{self.config.index_type}_index=1"
+            )
         self.client.command(self.schema)
 
+    def _schema(self, dim: int, index_params: Optional[Union[List, Dict]]) -> str:
+        """Create table schema
+        :param dim: dimension of embeddings
+        :param index_params: parameters used for index
+
+        This function returns a `CREATE TABLE` statement based on the value of
+        `self.config.index_type`.
+        If an index type is specified that index will be created, otherwise
+        no index will be created.
+        In the case of there being no index, a linear scan will be performed
+        when the embedding field is queried.
+        """
+
+        if self.config.index_type:
+            return f"""\
+        CREATE TABLE IF NOT EXISTS {self.config.database}.{self.config.table}(
+            {self.config.column_map['id']} Nullable(String),
+            {self.config.column_map['document']} Nullable(String),
+            {self.config.column_map['embedding']} Array(Float32),
+            {self.config.column_map['metadata']} JSON,
+            {self.config.column_map['uuid']} UUID DEFAULT generateUUIDv4(),
+            CONSTRAINT cons_vec_len CHECK length(
+                {self.config.column_map['embedding']}) = {dim},
+            INDEX vec_idx {self.config.column_map['embedding']} TYPE \
+        {self.config.index_type}({index_params}) GRANULARITY 1000
```

Comment:
@mneedham this seems wrong to me and happy to take your advice as clickhouse expert!

This will substitute in a python repr of a List/Dict, or "None" if `index_params` is None.

Maybe index params should be an optional str, and if it's none, we fill it out as "" instead of "None"?

Reply:
@efriis Yeah, I think you're right. I tried setting the line to the following:

> def _schema(self, dim: int, index_params: Optional[str]) -> str:

And now the linter is happy for me locally:

```
$ make lint_package
./scripts/check_pydantic.sh .
./scripts/lint_imports.sh
poetry run ruff .
[ "langchain_community" = "" ] || poetry run ruff format langchain_community --diff
1038 files already formatted
[ "langchain_community" = "" ] || poetry run ruff --select I langchain_community
[ "langchain_community" = "" ] || mkdir -p .mypy_cache && poetry run mypy langchain_community --cache-dir .mypy_cache
Success: no issues found in 1038 source files
```